### PR TITLE
Add ZTele

### PR DESCRIPTION
**Pull request by Hichatu**
Teleport back to spawn chat command

Teleports to a random spawnpoint after 5 seconds
Teleport is cancelled if you move too far from the initial position the command was fired

### CODE REVIEW

File: src/commands.cpp [hunk 1]
```diff
@@ -162,6 +165,89 @@ CON_COMMAND_CHAT(stopsound, "stop weapon sounds")
...
 }
 
+CON_COMMAND_CHAT(ztele, "teleport to spawn")
+{
+	if (!player)
+		return;
+
+	//Count spawnpoints (info_player_counterterrorist & info_player_terrorist)
+	SpawnPoint* spawn = nullptr;
+	CUtlVector<SpawnPoint*> spawns;
```
Comment (Poggicek) [inline src/commands.cpp:175]: vector is never deleted

File: src/commands.cpp [hunk 2]
```diff
@@ -162,6 +165,89 @@ CON_COMMAND_CHAT(stopsound, "stop weapon sounds")
...
+
+	//Pick and get position of random spawnpoint
+	int randomindex = rand() % spawns.Count()+1;
+	Vector spawnpos = spawns[randomindex]->GetAbsOrigin();
+
+	//Here's where the mess starts
+	CBasePlayerPawn* pPawn = player->m_hPawn();
+	if (pPawn)
+	{
+		if (pPawn->m_iHealth() > 0)
```
Comment (Poggicek) [inline src/commands.cpp:192]: Invert ifs to not have this staircase

File: src/commands.cpp [hunk 3]
```diff
@@ -31,6 +32,8 @@
 #include "adminsystem.h"
 
 #include "tier0/memdbgon.h"
+#include <ISmmPlugin.h>
```
Comment (Poggicek) [inline src/commands.cpp:35]: Is there a particular reason this is included?

**Review by Poggicek — DISMISSED** [review]

File: src/commands.cpp [hunk 4]
```diff
@@ -162,6 +165,89 @@ CON_COMMAND_CHAT(stopsound, "stop weapon sounds")
...
+	while (nullptr != (spawn = (SpawnPoint*)UTIL_FindEntityByClassname(spawn, "info_player_")))
+	{
+		if (spawn->m_bEnabled())
+		{
+			spawns.AddToTail(spawn);
+		}
+	}
+
+	//Pick and get position of random spawnpoint
+	int randomindex = rand() % spawns.Count()+1;
```
Comment (Vauff) [inline src/commands.cpp:185]: I've read that `srand()` needs to be called once per application run for proper randomness. This is already done in [@random targeting](https://github.com/Source2ZE/CS2Fixes/blob/ad0891501918bbd10c0c62f9af42cb657977a629/src/playermanager.cpp#L156-L160), but should probably be moved to plugin startup now that `rand()` is being used in several locations.

File: src/playermanager.cpp [hunk 5]
```diff
@@ -155,7 +155,6 @@ ETargetType CPlayerManager::TargetPlayerString(int iCommandClient, const char* t
 
 		if (!seeded)
 		{
-			srand(time(0));
 			seeded = true;
 		}
```
Comment (xen-000) [inline src/playermanager.cpp:159]: Remove this too and the declaration right above it

**Review by xen-000 — APPROVED** [review]

**Pull request merged**